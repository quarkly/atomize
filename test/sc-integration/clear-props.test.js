import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import atomize from '../../src';

const Element = atomize('div')({
  effects: { hover: ':hover' },
  aliases: true,
});

// eslint-disable-next-line react/jsx-props-no-spreading
const Child = props => <div {...props} />;

const ElementWithComponent = atomize(Child)({
  effects: { hover: ':hover' },
  aliases: true,
});

describe('atomize filter props keys', () => {
  test('filter', () => {
    const tree = renderer.create(<Element color="red" />).toJSON();
    expect(tree).toMatchInlineSnapshot(`
      .c0 {
        color: red;
      }

      <div
        className="c0"
      />
    `);
  });
  test('chains filter', () => {
    const tree = renderer.create(<Element hover-color="red" />).toJSON();
    expect(tree).toMatchInlineSnapshot(`
      .c0:hover,
      .c0[data-quarkly-state="hover"] {
        color: red;
      }

      <div
        className="c0"
      />
    `);
  });
  test('pass', () => {
    const tree = renderer.create(<Element color="red" color_passed="red" />).toJSON();
    expect(tree).toMatchInlineSnapshot(`
      .c0 {
        color: red;
      }

      <div
        className="c0"
        color_passed="red"
      />
    `);
  });
  test('pass to fn children', () => {
    const tree = renderer.create(<ElementWithComponent color="red" color_passed="red" />).toJSON();
    expect(tree).toMatchInlineSnapshot(`
      .c0 {
        color: red;
      }

      <div
        className="c0"
        color="red"
        color_passed="red"
      />
    `);
  });
  test('test props theme pass', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={{ breakpoints: { kek: [{ type: 'max-width', value: '1280' }] } }}>
          <ElementWithComponent kek-color="blue" color="red" color_passed="red" />
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchInlineSnapshot(`
      .c0 {
        color: red;
      }

      @media (max-width:1280px) {
        .c0 {
          color: blue;
        }
      }

      <div
        className="c0"
        color="red"
        color_passed="red"
        kek-color="blue"
      />
    `);
  });
  test('test props theme', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={{ breakpoints: { kek: [{ type: 'max-width', value: '1280' }] } }}>
          <Element kek-color="blue" color="red" color_passed="red" />
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchInlineSnapshot(`
      .c0 {
        color: red;
      }

      @media (max-width:1280px) {
        .c0 {
          color: blue;
        }
      }

      <div
        className="c0"
        color_passed="red"
      />
    `);
  });

  describe.each([undefined, false, true])(
    'test cleaning style props with forwardCssProperties = %p',
    forwardCssProperties => {
      const Primitive = atomize('div')({
        effects: { hover: ':hover' },
        aliases: true,
        forwardCssProperties,
      });

      const ComponentWithoutStylesProp = atomize(Child)({
        effects: { hover: ':hover' },
        aliases: true,
        forwardCssProperties,
      });

      test("isn't included styles", () => {
        const treeComponent = renderer
          .create(<ComponentWithoutStylesProp color="red" color_passed="red" />)
          .toJSON();
        const treePrimitive = renderer
          .create(<Primitive color="red" color_passed="red" />)
          .toJSON();

        expect(treeComponent).toMatchSnapshot();

        expect(treePrimitive).toMatchSnapshot();
      });
    },
  );
});
