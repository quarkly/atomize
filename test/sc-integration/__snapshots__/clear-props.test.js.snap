// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atomize filter props keys test cleaning style props with forwardCssProperties = false isn't included styles 1`] = `
.c0 {
  color: red;
}

<div
  className="c0"
  color_passed="red"
/>
`;

exports[`atomize filter props keys test cleaning style props with forwardCssProperties = false isn't included styles 2`] = `
.c0 {
  color: red;
}

<div
  className="c0"
  color_passed="red"
/>
`;

exports[`atomize filter props keys test cleaning style props with forwardCssProperties = true isn't included styles 1`] = `
.c0 {
  color: red;
}

<div
  className="c0"
  color="red"
  color_passed="red"
/>
`;

exports[`atomize filter props keys test cleaning style props with forwardCssProperties = true isn't included styles 2`] = `
.c0 {
  color: red;
}

<div
  className="c0"
  color="red"
  color_passed="red"
/>
`;

exports[`atomize filter props keys test cleaning style props with forwardCssProperties = undefined isn't included styles 1`] = `
.c0 {
  color: red;
}

<div
  className="c0"
  color="red"
  color_passed="red"
/>
`;

exports[`atomize filter props keys test cleaning style props with forwardCssProperties = undefined isn't included styles 2`] = `
.c0 {
  color: red;
}

<div
  className="c0"
  color_passed="red"
/>
`;
